### ISSUE TITLE
[BUG](vitest/no-done-callback): reports when using 

### ISSUE THREAD

**Issue by Haberkamp** [veritem/eslint-plugin-vitest#312]
Snapshot tests run in parallel need to use the local Test Context. To use the local Test Context I need to add a parameter to the callback function. But when I do that the `vitest/no-done-callback` throws an error. The rule assumes that I try to tell vitest that the test has ended. Which is not what I am actually doing.

```javascript
it.concurrent('some test name', ({ expect }) => {
                                  // ^ linting error appears here
  expect(true).toMatchSnapshot();
});
```
